src/plugins/kibana_utils/public/index.ts

```diff
@@ -17,4 +17,5 @@
  * under the License.
  */
 
+export * from './ajax_stream';
```

Comment:
The PR itself looks good, but if it's used only by interpreter, maybe it's best to put it inside interpreter.
(As long as we don't have a clear reason for other apps \ 3rd party apps using it).

Reply:
++ Agreed. I think we should have a bias toward _not_ moving stuff into a shared location unless it is clearly needed in multiple places and we are willing to support it as a public API.

It's always easy to move something into `kibana_utils` later and expose it for others to use, but much harder to put it here and then change our minds later, so I'd vote to keep it as an implementation detail of the interpreter if that's the only place using it now.

Reply:
I don't think I agree with both of you, because:

1. If it is a utility library it can perfectly live in a shared location, like many other utility libraries, regardless how often and where they are used.
1. `ajax_stream` was already in a shared location, this PR just moves it to a shared location in the New Platform.
1. And, no, it will actually not be used by Interpreter. Which is another reason we explicitly don't want it in Interpreter.
1. This lib is needed for Canvas. And in the future `bfetch` may need it.

Reply:
> regardless how often and where they are used. 

I disagree with this sentiment; I don't think we should expose something as shared code unless it actually _needs_ to be shared.

This is for the simple reason that 3rd party plugin devs might start using it without our awareness, and then we can't easily make changes to it as it is now a public API.

Part of what the new platform solves for us is the problem of having a lack of clear contracts. By putting things in a shared utils directory first, we risk recreating this issue similar to the `ui/public` situation where we have a huge file of "shared" code that might not actually need to be shared, then it isn't clear what is/isn't public.

> ajax_stream was already in a shared location

Personally i don't think it should have been, but that's just my opinion :) 

It may have been a result of the timing, I can't remember if this was developed when some pieces of the interpreter were still in Canvas or not

> This lib is needed for Canvas. And in the future bfetch may need it, too.

It is needed for Canvas for now because we are removing server stuff from the interpreter until we can design a long-term server side solution. But once we develop that solution, then it would live in the interpreter, which will be in the data plugin, just like bfetch. So long term I would still anticipate this moving into the data plugin, unless it becomes clear in the interim that it is needed across Kibana more widely. 

Ultimately I don't think it's the end of the world if we feel strongly that this must live in kibana_utils -- it's a relatively small piece of code after all -- but I want to be careful about using kibana_utils as a crutch just because it's convenient to put stuff in there.

Reply:
Agreed offline to move it to `interpreter/public/lib`.